### PR TITLE
update endpoints to support caching

### PR DESCRIPTION
**Pull request by spmistry**

### CODE REVIEW

File: crux/models/delivery.py [hunk 1]
```diff
@@ -103,3 +103,17 @@ def get_raw(self):
                 )
                 obj.refresh()
                 yield obj
+
+    def get_healthlog(self):
+        # type: () -> Iterator[Resource]
+        """Get the raw delivery data
```
Comment (nerdysquid) [inline crux/models/delivery.py:109]: Should be 
```suggestion
        """Get the delivery health log
```

File: crux/models/delivery.py [hunk 2]
```diff
@@ -103,3 +103,17 @@ def get_raw(self):
...
+
+        Returns:
+            list (:obj:`crux.models.Resource`): List of resources.
```
Comment (nerdysquid) [inline crux/models/delivery.py:112]: Doesn't return the list of resources

Comment (spmistry) [reply crux/models/delivery.py]: fixed

File: crux/models/delivery.py (hunk 1 above)
Comment (spmistry) [reply crux/models/delivery.py]: fixed

File: crux/models/dataset.py [hunk 3]
```diff
@@ -1159,6 +1169,7 @@ def get_files_range(
         self,
         start_date,  # type: Union[datetime, str]
         end_date=None,  # type: Optional[Union[datetime, str]]
+        delivery_status=None,  #type: str
```
Comment (stu-speed) [inline crux/models/dataset.py:1172]: This method returns files for "DELIVERY_SUCCEEDED" only so should not be a parameter but instead coded for that status only

Comment (spmistry) [reply crux/models/dataset.py]: updated to force status to succeeded when None.

**Pull request merged**